### ISSUE TITLE
tf.metrics doesn't include cross_entropy.

### ISSUE THREAD

**Issue by hanfeisun** [tensorflow/tensorflow#14041]
In tensorflow estimator, I want to use _cross entropy_ as the evaluation metrics (`eval_metric_ops` parameter of `EstimatorSpec`)

However, `tf.metrics` doesn't have this function. Also, tensorflow estimator doesn't allow me to use `tf.nn.sigmoid_cross_entropy_with_logits` as the `eval_metric_ops`.

**Comment by facaiy** [comment]
Let's make it clear, do you expect this?
![image](https://user-images.githubusercontent.com/1112263/32140251-b75b71be-bc94-11e7-9909-0a4ac3df906a.png)
where `p` is label, and `q` is prediction for binary classification.

**Comment by facaiy** [comment]
There is a slightly difference: `cross_entropy` expects `(labels, logits)`, while `tf.metrics` expects `(labels, predictions)`.

**Comment by hanfeisun** [comment]
@facaiy I see. But I was still a bit curious why there is no `cross_entropy` for `tf.metrics`..

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by jhseu** [comment]
Added the API review to consider whether adding cross entropy to tf.metrics makes sense.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by jhseu** [comment]
To clarify, why can't you use tf.metrics.mean here and pass in the result of tf.nn.sigmoid_cross_entropy_with_logits? It would effectively be the same thing.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by hanfeisun** [comment]
@jhseu Thanks, but I find tensorflow estimator doesn't allow me to use tf.nn.sigmoid_cross_entropy_with_logits as the eval_metric_ops..

**Comment by jhseu** [comment]
Yeah, but couldn't it be wrapped with tf.metrics.mean()?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
